pymtl3/passes/yosys/translation/structural/YosysStructuralTranslatorL2.py

```diff
@@ -217,7 +215,7 @@ def _gen_packed_array( dtype, n_dim, array ):
           assert False, "unrecognized data type {}!".format( dtype )
       else:
         ret = []
-        for i in reversed( range( n_dim[0] ) ):
+        for i in reversed( list(range( n_dim[0])) ):
```

Comment:
intermediate list again 😄 

(I'm going to go report a few issues against `modernize` later)